examples/demo-prover/host/src/main.rs

```diff
@@ -73,12 +73,14 @@ async fn main() -> Result<(), anyhow::Error> {
         demo.init_chain(genesis_config);
     }
 
-    demo.begin_slot(Default::default());
-    let (prev_state_root, _) = demo.end_slot();
-    let mut prev_state_root = prev_state_root.0;
+    let mut prev_state_root = {
+        let res = demo.apply_block(0, Default::default(), []);
+        res.state_root.0
+    };
 
     for height in rollup_config.start_height.. {
         let mut host = Risc0Host::new(ROLLUP_ELF);
+        host.write_to_guest(height);
```

Comment:
This seems risky. Let's compute it and store it in state instead. (I think Theo already has code to do this)

Reply:
Yes, I've removed it from this PR, because it is not necessary now.

It will be added when blobs deferring will start to be used